cloudflare.go

```diff
@@ -260,30 +260,26 @@ func (api *API) makeRequestWithAuthTypeAndHeaders(ctx context.Context, method, u
 		return nil, respErr
 	}
 
-	switch {
-	case resp.StatusCode >= http.StatusOK && resp.StatusCode < http.StatusMultipleChoices:
-	case resp.StatusCode == http.StatusUnauthorized:
-		return nil, errorFromResponse(resp.StatusCode, respBody)
-	case resp.StatusCode == http.StatusForbidden:
-		return nil, errorFromResponse(resp.StatusCode, respBody)
-	case resp.StatusCode == http.StatusServiceUnavailable,
-		resp.StatusCode == http.StatusBadGateway,
-		resp.StatusCode == http.StatusGatewayTimeout,
-		resp.StatusCode == 522,
-		resp.StatusCode == 523,
-		resp.StatusCode == 524:
-		return nil, errors.Errorf("HTTP status %d: service failure", resp.StatusCode)
-	// This isn't a great solution due to the way the `default` case is
-	// a catch all and that the `filters/validate-expr` returns a HTTP 400
-	// yet the clients need to use the HTTP body as a JSON string.
-	case resp.StatusCode == 400 && strings.HasSuffix(resp.Request.URL.Path, "/filters/validate-expr"):
-		return nil, errors.Errorf("%s", respBody)
-	default:
-		var s string
-		if respBody != nil {
-			s = string(respBody)
+	if resp.StatusCode >= http.StatusBadRequest {
+		if strings.HasSuffix(resp.Request.URL.Path, "/filters/validate-expr") {
+			return nil, errors.Errorf("%s", respBody)
+		}
+
+		if resp.StatusCode > http.StatusInternalServerError {
+			return nil, errors.Errorf("HTTP status %d: service failure", resp.StatusCode)
+		}
+
+		errBody := &Response{}
+		err = json.Unmarshal(respBody, &errBody)
+		if err != nil {
+			return nil, errors.Wrap(err, errUnmarshalErrorBody)
+		}
+
+		return nil, &APIRequestError{
+			StatusCode: resp.StatusCode,
+			Message:    errBody.Errors[0].Message,
```

Comment:
There are several API endpoints that output multiple error codes/messages in the single response. It would be great to expose them.

Reply:
sure thing, i've updated the usage to account for multiples in 0535f9b8616dba2f0baae5f70e877e4ab2b8c6db.